src/main/java/redis/clients/jedis/JedisFactory.java

```diff
@@ -18,50 +18,50 @@
 /**
  * PoolableObjectFactory custom impl.
  */
-class JedisFactory implements PooledObjectFactory<Jedis> {
+public class JedisFactory implements PooledObjectFactory<Jedis> {
   private final AtomicReference<HostAndPort> hostAndPort = new AtomicReference<>();
   private final int connectionTimeout;
   private final int soTimeout;
   private final int infiniteSoTimeout;
   private final String user;
-  private final String password;
+  private final AtomicReference<String> password = new AtomicReference<>();
```

Comment:
why do you need AtomicReference? volatile should do here, no?

Reply:
> why do you need AtomicReference?

because I have borrowed that piece of code from #1606 

> volatile should do here, no?

yes. both of those are same in java, isn't it?

Reply:
for get/set you're right but you just don't need it.

Reply:
addressed